File: app/src/main/java/org/bochman/upapp/domain/interactor/GetLastSearch.java
package org.bochman.upapp.domain.interactor;

import com.google.android.gms.maps.model.LatLng;

import org.bochman.upapp.data.repository.PoiRepository;

/**
 * Encapsulate a nearby search request.
 */
public class GetLastSearch {

    private final PoiRepository repo;

    GetLastSearch(PoiRepository repo){
        this.repo=repo;
    }

    String invoke(){
         return repo.loadSearchQuery();
    }

}

File: app/src/main/java/org/bochman/upapp/domain/interactor/GetLastSearch.kt
package org.bochman.upapp.domain.interactor;

import org.bochman.upapp.data.repository.PoiRepository

/**
 * Encapsulate a nearby search request.
 */
class GetLastSearch(private val poiRepository: PoiRepository) {

    suspend operator fun invoke(): String {
        return poiRepository.loadSearchQuery()
    }
}
